package.json

```diff
@@ -41,7 +41,8 @@
     "build:typechain:revocationregistryonchain": "typechain --target ethers-v5 --out-dir ethers './build/contracts/RevocationRegistryOnChain.json'",
     "deploy:stakingpoolfactory": "node -r ts-node/register ./scripts/deploy_staking_pool_factory",
     "deploy:revocationregistryfactory": "node -r ts-node/register ./scripts/deploy_revocationRegistry",
-    "deploy:roleDefResolver": "node -r ts-node/register ./scripts/deploy_roleDefResolver"
+    "deploy:roleDefResolver": "node -r ts-node/register ./scripts/deploy_roleDefResolver",
+    "preinstall": "npx npm-force-resolutions"
```

Comment:
Could you explain a bit more why we need `npm-force-resolutions`?
Is it because our dependencies themselves have dependencies which are at vulnerable versions and so we want to force an upgrade on them?

Reply:
Yeah, some of the nested dependencies are quite old and aren't updated in their respective packages / modules. So `npm-force-resolutions` is our last shot.

Reply:
Hmm, do you know which dependencies we have that are relying on these out of date nested dependencies?

Also, reading the `npm-force-resolutions`, it seems that if we go down this route then we'll have to manually maintain the `resolutions` packages ourselves going forward which could be cumbersome. So I'm a bit worried about going down this path. What are your thoughts on this?

Reply:
There are multiple dependencies which relies on a common dependency. For e.g ( `truffle`, `babel`, `eslint`, `ganache-cli`, `mocha`, `npm` and more ) has `ansi-regex` as its one of the nested dependencies with different versions. `eslint` has the latest versions but most of the nested dependencies are quite older than the vulnerable version reported in the security report (attached to the story). 

When we look at these nested dependencies, many are small modules which are not maintained / updated for more than a year or are deprecated but still being used in the parent dependencies (which are also not maintained at needed pace).

Agreed that `npm-force-resolutions` would put extra work to maintain. But this helps :
1. Prevents our repos from being vulnerable and react to security issues quickly rather than wanting the dependency maintainers to update their repos itself and publish.
2. For some dependencies we might not have alternative to switch to, so it's better to use them enforcing the latest versions to prevent vulnerability.
3. I tried to update some of dependencies with npm fixes, but it broke the code.

If we get better options / approach in future, we can always remove `npm-force-resolutions`.

Reply:
@nichonien could you explain a bit more the third point 
> I tried to update some of dependencies with npm fixes, but it broke the code.
To me, this is the preferable approach even if will take a little bit longer

Reply:
@jrhender Updating the vulnerabilities with `npm fix` actually doesn't upgrade all the dependencies ( mostly the nested ). Secondly `npm fix` bring in fixes that are published, so if A has a dependency B and the maintainers didn't upgrade B in the next release then it would still be vulnerable after the `npm fix`.